iac/terraform/aws/cloudfront/cloudfront.tf

```diff
@@ -3,29 +3,46 @@ resource "aws_cloudfront_distribution" "cdn" {
   #checkov:skip=CKV2_AWS_47: "CloudFront Distribution should have WAF enabled"
   #checkov:skip=CKV_AWS_86: "Ensure Cloudfront distribution has Access Logging enabled"
   #checkov:skip=CKV2_AWS_32: "Ensure CloudFront distribution has a response headers policy attached"
+
+  aliases             = [var.bloodconnect_environment_domain]
+  enabled             = true
+  is_ipv6_enabled     = false
+  comment             = "CloudFront distribution for front-end site"
+  default_root_object = "index.html"
+  price_class         = "PriceClass_100"
+
+  # S3 Primary Origin (Static Site)
   origin {
-    domain_name = aws_s3_bucket.static_site.bucket_regional_domain_name
+    domain_name = var.static_site_bucket.bucket_regional_domain_name
     origin_id   = var.cloudfront_distribution_origin_id
 
     s3_origin_config {
       origin_access_identity = aws_cloudfront_origin_access_identity.s3_static_bucket_oai.cloudfront_access_identity_path
     }
   }
 
+  # S3 Failover Origin
   origin {
-    domain_name = aws_s3_bucket.cloudfront_failover_bucket.bucket_regional_domain_name
+    domain_name = var.failover_bucket.bucket_regional_domain_name
     origin_id   = var.cloudfront_distribution_failover_origin_id
 
     s3_origin_config {
       origin_access_identity = aws_cloudfront_origin_access_identity.s3_static_bucket_oai.cloudfront_access_identity_path
     }
   }
 
-
-  enabled             = true
-  is_ipv6_enabled     = false
-  comment             = "CloudFront distribution for front-end site"
-  default_root_object = "index.html"
+  # API Gateway Origin (for /api routes)
+  origin {
+    domain_name = "${var.rest_api_id}.execute-api.${data.aws_region.current.name}.amazonaws.com"
```

Comment:
can't we get this from api gateway reference ?

Reply:
yes we can do it, but the URL is like `https://st06ivxx1j.execute-api.ap-south-1.amazonaws.com`. we'll have to update the string to remove the https. so this way seemed cleaner